nixos/modules/system/boot/loader/grub/install-grub.pl

```diff
@@ -370,6 +373,28 @@ sub GrubFs {
             fi
         ";
     }
+
+    rmtree("$bootPath/theme") or die "cannot clean up theme folder in $bootPath\n" if -e "$bootPath/theme";
+
+    if ($theme) {
+        # Copy theme
+        rcopy($theme, "$bootPath/theme") or die "cannot copy $theme to $bootPath\n";
+        $conf .= "
+            # Sets theme.
+            set theme=" . ($grubBoot->path eq "/" ? "" : $grubBoot->path) . "/theme/theme.txt
+            export theme
+            # Load theme fonts, if any
+         ";
+
+         find( { wanted => sub {
+             if ($_ =~ /\.pf2$/i) {
+                 $font = File::Spec->abs2rel($_, $theme);
+                 $conf .= "
+                     loadfont " . ($grubBoot->path eq "/" ? "" : $grubBoot->path) . "/theme/$font
+                 ";
+             }
```

Comment:
I'm not sure if you're currently working on it or not, but for the previously stated issue about non-top-level font files:


> > The wanted function takes no arguments but rather does its work through a collection of variables.
> > 
> >     `$File::Find::dir` is the current directory name,
> >     `$_` is the current filename within that directory
> >     `$File::Find::name` is the complete pathname to the file.
> > 
> 
> https://perldoc.perl.org/File/Find.html#The-wanted-function
> 
> I believe you should be able to replace `$_` with `$File::Find::name` to fix the issue with files in subfolders.